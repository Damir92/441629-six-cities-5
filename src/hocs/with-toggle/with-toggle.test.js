import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

import withToggle from './with-toggle';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <Fragment>
      {children}
    </Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withToggle(MockComponent);

describe(`with-auth-form hoc rendered correctly`, () => {
  it(`withToggle rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        toggleState={false}
      >
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
