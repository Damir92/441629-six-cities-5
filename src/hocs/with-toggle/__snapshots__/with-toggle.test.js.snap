// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-auth-form hoc rendered correctly withToggle rendered correctly 1`] = `null`;
